code/LeftAndMain.php

```diff
@@ -1910,18 +1507,25 @@ public function getSilverStripeNavigator()
      */
     public function currentPageID()
     {
+        if ($this->pageID) {
+            return $this->pageID;
+        }
         if ($this->getRequest()->requestVar('ID') && is_numeric($this->getRequest()->requestVar('ID'))) {
             return $this->getRequest()->requestVar('ID');
         } elseif ($this->getRequest()->requestVar('CMSMainCurrentPageID') && is_numeric($this->getRequest()->requestVar('CMSMainCurrentPageID'))) {
             // see GridFieldDetailForm::ItemEditForm
             return $this->getRequest()->requestVar('CMSMainCurrentPageID');
         } elseif (isset($this->urlParams['ID']) && is_numeric($this->urlParams['ID'])) {
             return $this->urlParams['ID'];
-        } elseif (Session::get($this->sessionNamespace() . ".currentPage")) {
+        }
+
+
+        /** @deprecated */
+        if (Session::get($this->sessionNamespace() . ".currentPage")) {
```

Comment:
Should we deprecate `CMSMain->sessionNamespace()` as well? AFAICT it's not used elsewhere.